src/main/java/org/mskcc/cmo/messaging/impl/JSGatewayImpl.java

```diff
@@ -0,0 +1,256 @@
+package org.mskcc.cmo.messaging.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Connection;
+import io.nats.client.Dispatcher;
+import io.nats.client.JetStream;
+import io.nats.client.JetStreamOptions;
+import io.nats.client.JetStreamSubscription;
+import io.nats.client.Message;
+import io.nats.client.Nats;
+import io.nats.client.Options;
+import io.nats.client.Options.Builder;
+import io.nats.client.api.PublishAck;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mskcc.cmo.common.FileUtil;
+import org.mskcc.cmo.common.impl.FileUtilImpl;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.mskcc.cmo.messaging.utils.SSLUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JSGatewayImpl implements Gateway {
+    private final Log LOG = LogFactory.getLog(JSGatewayImpl.class);
+    private final CountDownLatch publishingShutdownLatch = new CountDownLatch(1);
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final Map<String, JetStreamSubscription> subscribers = new HashMap<>();
+    private volatile BlockingQueue<PublishingQueueTask> publishingQueue = new LinkedBlockingQueue<>();
```

Comment:
I don't think that volatile is need here or for the Executor service as they get set once.